src/components/collabs/pick.svelte

```diff
@@ -27,11 +27,26 @@
 			return;
 		}
 
-		const response = _window.confirm($t('collabs.delete_pick_confirm'));
+		let confirmed: boolean = false;
+		let reason: string | null;
 
-		if (response) {
-			await axios.delete('/api/collabs/' + collab.id + '/picks/' + pick.id);
+		if ($discord?.admin && pick.userId !== $discord?.id) {
+			reason = _window.prompt('Whats the reason for the deletion?', 'Duplicate Pick');
+			if (reason) {
+				confirmed = true;
+			}
+		} else {
```

Comment:
This else still allows users that aren't an admin to call the delete API call if their user IDs don't match. Extract the admin check to where the prompt now is, and run call the prompt if the admin check passes. If it doesn't pass, make sure the API call does not get called.

Reply:
Replaced the else with if the user has the same id with the pick, it can be deleted.